packages/expo-modules-core/ios/Core/Views/SwiftUI/AutoSizingStack.swift

```diff
@@ -28,15 +28,19 @@ extension ExpoSwiftUI {
 
     public var body: some SwiftUI.View {
       if #available(iOS 16.0, tvOS 16.0, *) {
-        content.fixedSize(horizontal: axis.contains(.horizontal), vertical: axis.contains(.vertical))
-        .onGeometryChange(for: CGSize.self) { proxy in
-          proxy.size
-        } action: {
-          let width = axis.contains(.horizontal) ? $0.width : ShadowNodeProxy.UNDEFINED_SIZE
-          let height = axis.contains(.vertical) ? $0.height : ShadowNodeProxy.UNDEFINED_SIZE
-          let size = CGSize(width: width, height: height)
-          proxy.setViewSize?(size)
-        }
+        content
+          .if(proxy.setViewSize != nil, { view in
+            view
+              .fixedSize(horizontal: axis.contains(.horizontal), vertical: axis.contains(.vertical))
+              .onGeometryChange(for: CGSize.self) { proxy in
+                proxy.size
+              } action: {
+                let width = axis.contains(.horizontal) ? $0.width : ShadowNodeProxy.UNDEFINED_SIZE
+                let height = axis.contains(.vertical) ? $0.height : ShadowNodeProxy.UNDEFINED_SIZE
+                let size = CGSize(width: width, height: height)
+                proxy.setViewSize?(size)
+              }
+          })
```

Comment:
had to move this inside of a `.if()` because we only want to apply fixed size and the onGeometryChange if this is NOT a child of another swiftui view

Reply:
could we maybe make the entire `proxy` nullable? Imo that would be a clearer API if we ever add more fields to the proxy (like setting attrs, reading styling or anything like that)

Reply:
Unfortunately environmentObjects cannot be nullable, I tried this at first but the api won't allow it. 

> If not, let's try and export a proxy mock as a const and compare to that.

Can you clarify this a bit please?